pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -251,80 +250,93 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, nil
 	}
 
+	// Pick the action to perform
```

Comment:
We may need to add some provisioner APIs to probe some of these states. For example, if we provision a host and then the Pod is killed and restarted, we won't have it registered and will need to do that without reprovisioning it.